codec_test.go

```diff
@@ -111,13 +166,7 @@ func TestUnmarshalNumber(t *testing.T) {
 
 			var test Test
 			err := json.Unmarshal([]byte(s), &test)
-			if tc.wantErr != nil {
-				require.Equal(t, tc.wantErr, err)
-				return
-			}
-
-			require.NoError(t, err)
-			require.Equal(t, strings.Trim(tc.in, `"`), test.Test.String())
```

Comment:
why is this check removed? this ensures the data after unmarshaling matches the input and I think `TestUnmarshalText` should have a similar check

Reply:
The `require.ErrorIs(t, err, tc.wantErr)` check will fail for `nil` values if an error is expected and it's much more idiomatic to check the error instead of checking the string value.

Anyway if you want we can revert this change.

Reply:
I'm ok with `require.ErrorIs` but the test doesn't only check error. As I mentioned it also checks and makes sure the decimal value matches the string input

Reply:
@quagmt Sorry I misunderstood you, I'll resolve it now, I thought you were talking about the error check.